### PR TITLE
Bug 1168193 - Closing tab now animates out

### PR DESCRIPTION
**Pull request by sleroux**
Added the swipe animation to happen when the user taps the tab's X button. Also took the opportunity to clean up the SwipeAnimator code with some things I learned about transforms to simplify the animation code. Along with the simplification is a fix for the animation 'stopping' when user ends panning.

### CODE REVIEW

File: Client/Frontend/Browser/SwipeAnimator.swift [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+import Foundation
+
+struct SwipeAnimatorUX {
+    var totalRotationInDegrees: Double
+    var deleteThreshold: CGFloat
+    var totalScale: CGFloat
+    var totalAlpha: CGFloat
+    var minExitVelocity: CGFloat
+    var recenterAnimationDuration: NSTimeInterval
```
Comment (thebnich) [inline Client/Frontend/Browser/SwipeAnimator.swift]: Can these be `let`s?

File: Client/Frontend/Browser/SwipeAnimator.swift [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+private let DefaultSwipeAnimatorUX =
+    SwipeAnimatorUX(
```
Comment (thebnich) [inline Client/Frontend/Browser/SwipeAnimator.swift]: Do we plan on having multiple `SwipeAnimatorUX`s? This abstraction seems like overkill...I'd just bake these constants directly into the `struct` like we do in other places, unless we have a good reason not to.

File: Client/Frontend/Browser/SwipeAnimator.swift [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+    let container: UIView
+    let animatingView: UIView
+    private let ux: SwipeAnimatorUX
+
+    var containerCenter: CGPoint {
+        return CGPoint(x: CGRectGetWidth(container.frame) / 2, y: CGRectGetHeight(container.frame) / 2)
+    }
+
+    init(animatingView: UIView, container: UIView, ux: SwipeAnimatorUX = DefaultSwipeAnimatorUX) {
```
Comment (thebnich) [inline Client/Frontend/Browser/SwipeAnimator.swift]: It's also strange that we're passing the `SwipeAnimatorUX` into the `init` here. I think we made these `UX` `struct`s just as a way to collect constants into a single place, with maybe the eventual goal that they'll be going into some shared constants file.

I guess you're treating `SwipeAnimatorUX` as a kind of skin?

File: Client/Frontend/Browser/SwipeAnimator.swift (hunk 1 above)
Comment (sleroux) [inline Client/Frontend/Browser/SwipeAnimator.swift]: Yup I can change these over

File: Client/Frontend/Browser/SwipeAnimator.swift (hunk 2 above)
Comment (sleroux) [inline Client/Frontend/Browser/SwipeAnimator.swift]: Originally no, but I have a ticket for deleting tiles off the home panel where we're going to be using the swipe animator to swipe off the tiles and the thresholds will be different there. I think renaming this to something not UX would make more sense so it doesn't get mixed up with the eventual app constants.

File: Client/Frontend/Browser/SwipeAnimator.swift (hunk 3 above)
Comment (sleroux) [inline Client/Frontend/Browser/SwipeAnimator.swift]: See comment above

**Comment by sleroux** [comment]
@thebnich Just wanted to double check - are you okay with the changes I made to renaming the UX structure?

**Comment by thebnich** [comment]
Ah, I hadn't even noticed that these changes were going into a new `SwipeAnimator` file. This makes a lot more sense looking at it a second time. Looks good to me!

**Pull request merged**